storage/boltz/indexes.go

```diff
@@ -87,14 +99,38 @@ func (indexer *Indexer) addFkIndex(symbol EntitySymbol, fkSymbol EntitySetSymbol
 
 	indexer.addConstraint(index)
 	fkStore := fkSymbol.GetStore()
-	if baseStore, ok := fkStore.(*BaseStore); ok {
+	if baseStore, ok := fkStore.(constrained); ok {
 		baseStore.addConstraint(&fkDeleteConstraint{
 			symbol:   fkSymbol,
 			fkSymbol: symbol,
 		})
 	} else {
-		pfxlog.Logger().Warnf("fk store %v is not an indexer, can't enforce validity of constraint on delete",
-			fkSymbol.GetStore().GetEntityType())
+		panic(errors.Errorf("linked store %v is not constrained, can't enforce validity of constraint on delete",
```

Comment:
`pfxlog.Logger().Panic()` ?

Reply:
It'll only happen on startup and only if there's a coding error, so seemed reasonable.